internal/backends/collection_test.go

```diff
@@ -197,6 +199,78 @@ func TestCollectionInsertAllQueryExplain(t *testing.T) {
 	}
 }
 
+func TestCappedCollectionInsertAllDeleteAll(t *testing.T) {
+	t.Parallel()
+
+	ctx := conninfo.Ctx(testutil.Ctx(t), conninfo.New())
+
+	for name, b := range testBackends(t) {
+		name, b := name, b
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			dbName := testutil.DatabaseName(t)
+			collName := testutil.CollectionName(t)
+
+			db, err := b.Database(dbName)
+			require.NoError(t, err)
+
+			err = db.CreateCollection(ctx, &backends.CreateCollectionParams{
+				Name:       collName,
+				CappedSize: 8192,
+			})
+			require.NoError(t, err)
+
+			coll, err := db.Collection(collName)
+			require.NoError(t, err)
+
+			doc1 := must.NotFail(types.NewDocument("_id", int32(1)))
+			doc1.SetRecordID(1)
+
+			docMax := must.NotFail(types.NewDocument("_id", int32(2)))
+			docMax.SetRecordID(math.MaxInt64)
+
+			docMaxUint := must.NotFail(types.NewDocument("_id", int32(3)))
+			docMaxUint.SetRecordID(2*math.MaxInt64 + 1) // see type uint64 godoc
```

Comment:
Still investigating this part - how it's stored, how we should store it instead etc.

Reply:
> // see type uint64 godoc

Well, okay… https://pkg.go.dev/builtin@go1.21.4#uint64

> uint64 is the set of all unsigned 64-bit integers. Range: 0 through 18446744073709551615.

Not sure how that helps. And why don't we use `MaxUint64`?